src/Compilers/Core/Portable/Syntax/SyntaxToken.cs

```diff
@@ -78,7 +78,7 @@ private string GetDebuggerDisplay()
 
         internal int Index { get; }
 
-        internal int Position { get; }
+        public int Position { get; }
```

Comment:
let's break this out into a separate PR.  It will be more complex to do this since it is part of a public API surface and is in the compiler layer.

Reply:
Retested with this particular change off, seems not to affect the speedup too much (speedup in top post). So will revert it entirely.